src/app/cart.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { Item } from './models/item';
+import { BehaviorSubject, Observable, map } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CartService {
+  private cart = new Map<Item, number>();
+  private cartSubject = new BehaviorSubject<Map<Item, number>>(new Map());
```

Comment:
Might have been easier to create a new type for `CartItem` that had the quantity and the item data together. That way you wouldn't have to maintain two maps